src/widgets/list.rs

```diff
@@ -156,6 +156,72 @@ impl ListState {
             self.offset = 0;
         }
     }
+
+    /// Selects the next item or the first one if no item is selected
+    ///
+    /// Note: until the list is rendered, the number of items is not known, so the index is set to
+    /// `0` and will be corrected when the list is rendered
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// let mut state = ListState::default();
+    /// state.select_next();
+    /// ```
+    pub fn select_next(&mut self) {
+        let next = self.selected.map_or(0, |i| i.saturating_add(1));
```

Comment:
It might be a good idea to add support for wrap around here and on prev, but for now I left that out of the PR.

Reply:
On my tree widgets I found wrap around to be a bit more confusing. Hitting keys with nothing selected selects something here. Up ends at the top. For someone to move to the extremes, Home and End seem way better.

Reply:
There are a variety of use cases where this is reasonable. What I meant was that this should be something that can be enabled by an app rather than something that needs to be part of this PR.